cmake/BundledAzureSdkCpp.cmake

```
@@ -0,0 +1,126 @@
+        set(AZURE_IDENTITY_LIB "${BINARY_DIR}/thirdparty/azure-sdk-cpp-src/sdk/identity/azure-identity/${PREFIX}azure-identity.${SUFFIX}")
+    endif()
+
+    set(AZURESDK_LIBRARIES_LIST
+            "${AZURE_CORE_LIB}"
+            "${AZURE_STORAGE_COMMON_LIB}"
+            "${AZURE_STORAGE_BLOBS_LIB}"
+            "${AZURE_IDENTITY_LIB}")
+
+    set(AZURE_SDK_CMAKE_ARGS ${PASSTHROUGH_CMAKE_ARGS})
+    append_third_party_passthrough_args(AZURE_SDK_CMAKE_ARGS "${AZURE_SDK_CMAKE_ARGS}")
+
+    # Build project
+    ExternalProject_Add(
+            azure-sdk-cpp-external
+            GIT_REPOSITORY "https://github.com/Azure/azure-sdk-for-cpp.git"
+            GIT_TAG "azure-storage-blobs_12.0.0-beta.6"
+            BUILD_IN_SOURCE true
+            SOURCE_DIR "${BINARY_DIR}/thirdparty/azure-sdk-cpp-src"
+            BUILD_BYPRODUCTS "${AZURESDK_LIBRARIES_LIST}"
+            EXCLUDE_FROM_ALL TRUE
+            STEP_TARGETS build
+            CMAKE_ARGS ${AZURE_SDK_CMAKE_ARGS}
+            LIST_SEPARATOR % # This is needed for passing semicolon-separated lists
+            DEPENDS libxml2-external
+    )
+
+    # Set dependencies
+    add_dependencies(azure-sdk-cpp-external LibXml2::LibXml2 CURL::libcurl OpenSSL::Crypto OpenSSL::SSL nlohmann_json::nlohmann_json)
+
+    # Set variables
+    set(LIBAZURE_FOUND "YES" CACHE STRING "" FORCE)
+    set(LIBAZURE_INCLUDE_DIRS
+            "${BINARY_DIR}/thirdparty/azure-sdk-cpp-src/sdk/core/azure-core/inc/"
+            "${BINARY_DIR}/thirdparty/azure-sdk-cpp-src/sdk/storage/azure-storage-blobs/inc/"
+            "${BINARY_DIR}/thirdparty/azure-sdk-cpp-src/sdk/storage/azure-storage-common/inc/"
+            "${BINARY_DIR}/thirdparty/azure-sdk-cpp-src/sdk/identity/azure-identity/inc/"
+            CACHE STRING "" FORCE)
+    set(LIBAZURE_LIBRARIES ${AZURESDK_LIBRARIES_LIST} CACHE STRING "" FORCE)
+
+    # Create imported targets
+    FOREACH(LIBAZURE_INCLUDE_DIR ${LIBAZURE_INCLUDE_DIRS})
+        file(MAKE_DIRECTORY ${LIBAZURE_INCLUDE_DIR})
+    ENDFOREACH(LIBAZURE_INCLUDE_DIR)
+
+    add_library(AZURE::azure-core STATIC IMPORTED)
+    set_target_properties(AZURE::azure-core PROPERTIES IMPORTED_LOCATION "${AZURE_CORE_LIB}")
+    add_dependencies(AZURE::azure-core azure-sdk-cpp-external-build)
+    set_property(TARGET AZURE::azure-core APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${LIBAZURE_INCLUDE_DIRS})
+    set_property(TARGET AZURE::azure-core APPEND PROPERTY INTERFACE_LINK_LIBRARIES LibXml2::LibXml2 CURL::libcurl OpenSSL::Crypto OpenSSL::SSL Threads::Threads nlohmann_json::nlohmann_json)
+    if (APPLE)
+        set_property(TARGET AZURE::azure-core APPEND PROPERTY INTERFACE_LINK_LIBRARIES "-framework CoreFoundation")
+    endif()
+    if (WIN32)
+        set_property(TARGET AZURE::azure-core APPEND PROPERTY INTERFACE_LINK_LIBRARIES winhttp.lib)
+    endif()
```

Instead of setting properties directly, we should prefer using target commands to set them indirectly. This way is shorter, looks better and is more commonly found in examples and docs.
```suggestion
    target_include_directories(AZURE::azure-core INTERFACE ${LIBAZURE_INCLUDE_DIRS})
    target_link_libraries(AZURE::azure-core INTERFACE LibXml2::LibXml2 CURL::libcurl OpenSSL::Crypto OpenSSL::SSL Threads::Threads nlohmann_json::nlohmann_json)
    if (APPLE)
        target_link_libraries(AZURE::azure-core INTERFACE "-framework CoreFoundation")
    endif()
    if (WIN32)
        target_link_libraries(AZURE::azure-core INTERFACE winhttp.lib)
    endif()
```

---

Updated in [3dba6a8](https://github.com/apache/nifi-minifi-cpp/pull/979/commits/3dba6a8440d962cd546a7975e4fa9b21386ab6d0)